### ISSUE TITLE
undefined behaviour, out of bounds in GLZ_SKIP_WS

### ISSUE THREAD

**Issue by pauldreik** [stephenberry/glaze#1167]
The following test added to reflection_test.cpp:

```cpp
    "reflect_fuzz_fail"_test = [] {
       std::string buffer = "\x9e";
       my_struct obj{};
       expect(!glz::read_json(obj, buffer));
    };
```

fails when running under sanitizers:

> 10:17:00: Starting xxxxxxxxxxxxxxxxx/glaze/out/build/clang_latest/tests/reflection_test/reflection_test...
> /usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/array:61:33: runtime error: index 18446744073709551518 out of bounds for type 'const _Type' (aka 'const bool[256]')
> SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/array:61:33 
> /usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/array:61:33: runtime error: addition of unsigned offset to 0x55fbae531f20 overflowed to 0x55fbae531ebe
> SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/array:61:33 
> FAILED "reflect_write

found in main, commit eb7eb4083dcefa5f04139db1f793411bb920f15e

**Comment by stephenberry** [comment]
Thanks for reporting this! I'm working on improvements to string reading/writing right now in #1168 and will also look into this.

**Comment by stephenberry** [comment]
Oh, I see this isn't an issue with string reading, but whitespace handling for invalid inputs.

**Comment by stephenberry** [comment]
The issue was not converting signed characters to `uint8_t` before looking up indexes in tables. This revealed a number of places for potential address issues that have now been fixed in #1168

**Comment by stephenberry** [comment]
Note that in order for your test to be correct it should be: `expect(glz::read_json(obj, buffer));`. Remove the `not` because an error code is expected since this is invalid input for a JSON object.

But, thanks so much for revealing this issue!
